### PR TITLE
Ignore LLVM ABI in dlltool tests since those targets don't use dlltool

### PR DESCRIPTION
**Pull request by mati865**
Otherwise those two tests fail when running `./x.py test` with this target.

### CODE REVIEW

**Comment by bjorn3** [comment]
`//@ ignore-llvm` reads to me like it ignores the test when using the LLVM codegen backend, not when using the `*-gnullvm` targets.

**Comment by mati865** [comment]
I agree it's ambiguous but also in line with other ABIs. I'm open for suggestions though.

Personally I'd love if `ignore` statements were more precise, like `//@ignore-abi llvm` or `//@ignore-os windows`.

**Comment by jieyouxu** [comment]
I feel like we should fix needs-dlltools instead, like, what's the point of needs-dlltools if the test still runs on a target without dlltools?

**Comment by jeremyd2019** [comment]
`needs-dlltool` seems to answer the question "is dlltool on the PATH", which is a different question than "will the target actually try to *use* dlltool".  For gnullvm, it will not try to use dlltool, so the fact that there is one on the PATH is not particularly relevant.

For reference, the condition that decides whether to use dlltool or use llvm APIs directly to generate an import library is https://github.com/rust-lang/rust/blob/f9b16149208c8a8a349c32813312716f6603eb6f/compiler/rustc_codegen_llvm/src/common.rs#L387

**Comment by fee1-dead** [comment]
r? compiler

**Comment by jeremyd2019** [comment]
It looks like `tests\run-make\raw-dylib-custom-dlltool` fails similarly

**Comment by davidtwco** [comment]
> I feel like we should fix needs-dlltool instead, like, what's the point of needs-dlltool if the test still runs on a target without dlltools?

I'd agree with this, we should extend `needs-dlltool` so that it isn't just "there is a dlltool" but "the target will use it", `ignore-llvm` seems like it is ignoring that codegen backend.

**Comment by mati865** [comment]
Tested only x86_64-pc-windows-gnullvm but the changes are simple enough.

File: tests/run-make/raw-dylib-cross-compilation/Makefile [hunk 1]
```diff
@@ -1,8 +1,7 @@
 # Tests that raw-dylib cross compilation works correctly
 
 # only-gnu
```
Comment (jeremyd2019) [inline tests/run-make/raw-dylib-cross-compilation/Makefile:3]: Probably silly, but was there a reason this `only-gnu` was left in place while the other (.rs) tests removed theirs?

Comment (mati865) [reply tests/run-make/raw-dylib-cross-compilation/Makefile]: It was an oversight.

**Review by davidtwco — APPROVED** [review]

**Pull request merged**